### PR TITLE
[TAS-7] Miscellaneous improvements

### PR DESCRIPTION
**Pull request by saulprl**
## Describe your changes

In this PR, I cleaned up some issues with the currently available components to close the gap between between the app and the Figma design.

## What was done in this pull request

- [x] Added date formatting with `date-fns` ([docs](https://date-fns.org/v3.6.0/docs/Getting-Started)) for the `EventCard` component.
- [x] Added missing props and data to `EventCard`.
- [x] Adjust `EventCard` styles, mainly to keep consistent sizes.
- [x] Added our CMS to Next.js' config for image optimization.
- [x] Introduce a `BrandingHeader` component with default styles for consistency.
- [x] Added a `SectionTitle` component.
- [x] Fixed the CSI PRO Logo size in the `Navbar` component.
- [x] Added `zod` ([docs](https://zod.dev/)) for better type safety via `schemas`.

### Before

image

### After

image

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,29 +1,44 @@
...
+import {
+  createResponseSchema,
+  generateEmptyResponse,
+} from "@/models/cms-response";
+import { Event } from "@/models/events";
 
 const fetchEvents = async () => {
   const eventsRes = await fetch(
     "https://admin.csipro.isi.unison.mx/api/eventos",
+    { cache: "no-store" },
```
Comment (saulprl) [inline src/app/page.tsx:14]: ?

**Comment by Kartleth** [comment]
LGTM 👍

**Review by karolinaconk — APPROVED** [review]
LGTM

**Review by Kartleth — APPROVED** [review]
LGTM TOO

**Review by LUIS021201 — APPROVED** [review]
al putazo todo

**Pull request merged**